File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/modules/notices/Notice.java
package pl.szczodrzynski.edziennik.data.db.modules.notices;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.Index;

@Entity(tableName = "notices",
        primaryKeys = {"profileId", "noticeId"},
        indices = {@Index(value = {"profileId"})})
public class Notice {
    public int profileId;

    @ColumnInfo(name = "noticeId")
    public long id;

    @ColumnInfo(name = "noticeText")
    public String text;
    @ColumnInfo(name = "noticeSemester")
    public int semester;
    public static final int TYPE_NEUTRAL = 0;
    public static final int TYPE_POSITIVE = 1;
    public static final int TYPE_NEGATIVE = 2;
    @ColumnInfo(name = "noticeType")
    public int type = TYPE_NEUTRAL;

    public float points = 0;
    public String category = null;

    public long teacherId;

    @Ignore
    public Notice() {}

    public Notice(int profileId, long id, String text, int semester, int type, long teacherId) {
        this.profileId = profileId;
        this.id = id;
        this.text = text;
        this.semester = semester;
        this.type = type;
        this.teacherId = teacherId;
    }
}

File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/entity/Notice.kt
/*
 * Copyright (c) Kuba Szczodrzyński 2020-4-25.
 */
package pl.szczodrzynski.edziennik.data.db.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.Index

@Entity(tableName = "notices",
        primaryKeys = ["profileId", "noticeId"],
        indices = [
            Index(value = ["profileId"])
        ])
open class Notice(
        val profileId: Int,
        @ColumnInfo(name = "noticeId")
        var id: Long,
        @ColumnInfo(name = "noticeType")
        var type: Int,
        @ColumnInfo(name = "noticeSemester")
        var semester: Int,

        @ColumnInfo(name = "noticeText")
        var text: String,
        @ColumnInfo(name = "noticeCategory")
        var category: String?,
        @ColumnInfo(name = "noticePoints")
        var points: Float?,

        var teacherId: Long,
        var addedDate: Long = System.currentTimeMillis()
) : Keepable(), Noteable {
    companion object {
        const val TYPE_NEUTRAL = 0
        const val TYPE_POSITIVE = 1
        const val TYPE_NEGATIVE = 2
    }

    @Ignore
    var showAsUnseen = false

    override fun getNoteType() = Note.OwnerType.BEHAVIOR
}
